docs/docs/lists-and-keys.md

```diff
@@ -130,7 +130,7 @@ const todoItems = todos.map((todo, index) =>
 );
 ```
 
-We don't recommend using indexes for keys if the items can reorder, as that would be slow. You may read an [in-depth explanation about why keys are necessary](/react/docs/reconciliation.html#recursing-on-children) if you're interested.
+We don't recommend using indexes for keys if the items can reorder, as that would impact performance and may cause issues with the components state. If you choose not to assign a key to your list items then React will use indexes as keys. You may read an [in-depth explanation about why keys are necessary](/react/docs/reconciliation.html#recursing-on-children) if you're interested in more information.
```

Comment:
I've added a little more details here but I would hope that if someone does want more information that they will click through to the `in-depth explanation...`  contained in `/react/docs/reconciliation.html#recursing-on-children` that I have also updated.